### DISCUSSION TITLE
[question] Hide results if there is no cache with given input

### DISCUSSION THREAD

**Discussion by ojczeo** [Akryum/vue-apollo#258] [Q&A]
@Akryum Is there any way to load cached query to component only when cache exists with current params?
Flow which I want to achive:
I have `query` animals with variable `name`
1. I'm searching for dogs - apollo loads animals query from api and display dogs
2. I'm searching for cats - apollo hides dogs, next displaying cats results from api
3.  I'm searching for dogs again - apollo is loading dogs from cache first, next it displays new result from api.

For now when I'm using `cache-and-network` fetch policy I would like to avoid displaying caches when query is loading unless cache with exactly the same variables exists. 
@Akryum how to achieve that?

**Comment by jpikora** [comment]
@ojczeo you can hide the list while `loading` is truthy (>0). If the response is served from cache `loading` will stay falsy (0).

**Comment by ojczeo** [comment]
@jpikora Thanks, I already know it, but this is not a solution which I'm asking - I would like to **display** **cached** response when **loading response params match the cached response params** - if there is not match, not display. Your solution hide cached queries during each request.

**Comment by Akryum** [comment]
@ojczeo You have nothing special to do with `cache-and-network`, apollo will immediately give the cached result if any and update it later when the server response arrives.

**Comment by kestred** [comment]
@Akryum I think I have a very similar question (/ feature request?) which sounds almost the same as ojczeo's --- and that is that:

Currently, `$apollo.loading` and `$apollo.query.myQuery.loading` are `true` when using `cache-and-network` while the network request is still active.

I'd like to use something _like_ `$apollo.loading` in the Vue template to control whether the loading state is rendered, __however__ I want to eagerly render the data immediately if it is available from the cache (without hand coding the "I have data" condition for each view).

I think ideally there might be something like an `$apollo.eager` or `$apollo.resolved` or similar that is `true` for SmartQueries that were fetched from the catch (even while a network request is in progress), but `false` if there was no cache until the network request completes.
